pkg/module_manager/models/hooks/module_hook.go

```diff
@@ -0,0 +1,173 @@
+package hooks
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	types2 "github.com/flant/addon-operator/pkg/hook/types"
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/pkg/module_manager/models/hooks/kind"
+	"github.com/flant/shell-operator/pkg/hook/controller"
+	"github.com/flant/shell-operator/pkg/hook/types"
+)
+
+// ModuleHook hook which belongs to some module
+type ModuleHook struct {
+	executableHook
+	config *ModuleHookConfig
+}
+
+// NewModuleHook build new hook for a module
+//
+//	ex - some kind of executable hook (GoHook or ShellHook)
+func NewModuleHook(ex executableHook) *ModuleHook {
+	return &ModuleHook{
+		executableHook: ex,
+		config:         &ModuleHookConfig{},
+	}
+}
+
+// GetConfigVersion returns config version
+func (mh *ModuleHook) GetConfigVersion() string {
+	return mh.config.Version
+}
+
+// GetHookConfig returns config for the module hook, it has some difference with global hook
+func (mh *ModuleHook) GetHookConfig() *ModuleHookConfig {
+	return mh.config
+}
+
+// Order returns hook order
+func (mh *ModuleHook) Order(binding types.BindingType) float64 {
+	if mh.config.HasBinding(binding) {
+		switch binding {
+		case types.OnStartup:
+			return mh.config.OnStartup.Order
+		case types2.BeforeHelm:
+			return mh.config.BeforeHelm.Order
+		case types2.AfterHelm:
+			return mh.config.AfterHelm.Order
+		case types2.AfterDeleteHelm:
+			return mh.config.AfterDeleteHelm.Order
+		}
+	}
+	return 0.0
+}
+
+// InitializeHookConfig initializes the global hook config
+// for GoHook config is precompiled, so we just have to fetch it
+// for ShellHook, that hook will be run with `--config` flag, returns and parses the config
+func (mh *ModuleHook) InitializeHookConfig() (err error) {
+	switch hk := mh.executableHook.(type) {
+	case *kind.GoHook:
+		cfg := hk.GetConfig()
+		err := mh.config.LoadAndValidateGoConfig(cfg)
+		if err != nil {
+			return err
+		}
+
+	case *kind.ShellHook:
+		cfg, err := hk.GetConfig()
+		if err != nil {
+			return err
+		}
+		err = mh.config.LoadAndValidateShellConfig(cfg)
+		if err != nil {
+			return err
+		}
+
+	default:
+		return fmt.Errorf("unknown hook kind: %s", reflect.TypeOf(hk))
+	}
+
+	// Make HookController and GetConfigDescription work.
+	mh.executableHook.BackportHookConfig(&mh.config.HookConfig)
+
+	return nil
+}
+
+// SynchronizationNeeded is true if there is binding with executeHookOnSynchronization.
+func (mh *ModuleHook) SynchronizationNeeded() bool {
+	for _, kubeBinding := range mh.config.OnKubernetesEvents {
+		if kubeBinding.ExecuteHookOnSynchronization {
```

Comment:
This part of code was just removed. Seems its some inner representation of a config, because it's not a pointer here
<img width="588" alt="image" src="https://github.com/flant/addon-operator/assets/4477098/b793d5bb-57a3-4c1d-8417-c19b4478f2ef">